tests/Integration/PrestaShopBundle/Controller/GridControllerTestCase.php

```diff
@@ -44,188 +44,108 @@ abstract class GridControllerTestCase extends WebTestCase
     protected $client;
 
     /**
-     * Route to the grid you are testing
-     *
-     * @var string
-     */
-    protected $gridRoute;
-
-    /**
-     * The id of the test entity with which filters will be tested
-     * Should be set during SetUp
-     *
-     * @var int
-     */
-    protected $testEntityId;
-
-    /**
-     * The name of entity you are testing, e.g.,address.
-     *
-     * @var string
-     */
-    protected $testEntityName;
-
-    /**
-     * The route to create entity
-     *
-     * @var string
-     */
-    protected $createEntityRoute;
-
-    /**
-     * The route to delete entity
-     *
-     * @var string
-     */
-    protected $deleteEntityRoute;
-
-    /**
-     * Amount of entities in starting list
-     *
-     * @var int
+     * @var RouterInterface
      */
-    protected $initialEntityCount;
+    protected $router;
 
     /**
      * @var FormFiller
      */
     protected $formFiller;
 
-    /**
-     * Service id form form handler
-     *
-     * @var string
-     */
-    protected $formHandlerServiceId;
-
-    /**
-     * Save button id should always be the same, but can be overridden if needed
-     *
-     * @var string
-     */
-    protected $saveButtonId = 'save-button';
-
-    public function __construct($name = null, array $data = [], $dataName = '')
-    {
-        parent::__construct($name, $data, $dataName);
-
-        $this->formFiller = new FormFiller();
-    }
-
-    /**
-     * Creates a test entity and ensures asserts that amount of entities in the list got increased by one
-     *
-     * @throws TypeException
-     */
     public function setUp(): void
     {
         $this->client = static::createClient();
-        $this->client->followRedirects(true);
-
-        /** Asserts that list contains as many entities as expected */
-        $crawler = $this->client->request('GET', $this->getIndexRoute($this->client->getKernel()->getContainer()->get('router')));
-        $this->assertResponseIsSuccessful();
-
-        $entities = $this->getEntityList($crawler);
-        $this->initialEntityCount = $entities->count();
-
-        $this->createTestEntity();
-
-        /** Asserts amount of entities in the list increased by one and test entity exists */
-        $crawler = $this->client->request('GET', $this->getIndexRoute($this->client->getKernel()->getContainer()->get('router')));
-        $entities = $this->getEntityList($crawler);
-        /* If this fails it means entity was not created correctly */
-        self::assertCount($this->initialEntityCount + 1, $entities);
-        $this->assertTestEntityExists($entities);
+        $this->router = $this->client->getContainer()->get('router');
+        $this->formFiller = new FormFiller();
     }
 
     /**
-     * Removes the created test entity and asserts that it was successfully removed from the list.
+     * Calls the gris page and return the parsed entities it contains, based on the parseEntityFromRow that each
```

Comment:
A little typo here, I think.
```suggestion
     * Calls the grids page and return the parsed entities it contains, based on the parseEntityFromRow that each
```

Reply:
Nice catch, thanks!